src/App.tsx

```diff
@@ -1,34 +1,150 @@
+    query: '',
+    filter: FilterStatus.all,
+    selectedTodoId: null,
+    selectedUserId: null,
+  };
+
+  async componentDidMount() {
+    const todos = await api.getTodos();
+
+    this.setState({ todos });
+  }
+
+  loadUser = async () => {
+    if (this.state.selectedUserId !== null) {
+      // eslint-disable-next-line
+      const user = await api.getUser(this.state.selectedUserId);
+
+      this.setState({ user });
+    }
+  };
+
+  render() {
+    const {
+      todos,
+      user,
+      query,
+      filter,
+      selectedTodoId,
+      selectedUserId,
+    } = this.state;
+    const lowQuery = query.toLowerCase().trim();
+    let visibleTodos = [...todos];
+
+    const handleChangeQuery = (event: ChangeEvent<HTMLInputElement>) => {
+      return this.setState({ query: event.target.value });
+    };
 
-export const App: React.FC = () => {
-  return (
-    <>
+    const handleQueryReset = () => {
+      return this.setState({ query: '' });
+    };
+
+    const handleFilterChange = (event: ChangeEvent<HTMLSelectElement>) => {
+      return this.setState({ filter: event.target.value as FilterStatus });
+    };
+
+    const changeTodo = (todoId: number | null) => {
+      this.setState({ selectedTodoId: todoId });
+    };
+
+    const changeUser = (userId: number | null) => {
+      this.setState({ selectedUserId: userId });
+    };
+
+    const resetUser = () => {
+      this.setState({ user: null });
+    };
+
+    if (query[0] === ' ') {
+      this.setState({ query: '' });
+    }
+
+    if (filter === FilterStatus.active) {
+      visibleTodos = todos.filter(
+        (todo: Todo) => todo.completed === false,
+      );
+    }
+
+    if (filter === FilterStatus.completed) {
+      visibleTodos = todos.filter(
+        (todo: Todo) => todo.completed === true,
+      );
+    }
+
+    visibleTodos = visibleTodos.filter(
+      (todo: Todo) => todo.title.toLocaleLowerCase().includes(lowQuery),
+    );
+
+    return (
       <div className="section">
         <div className="container">
           <div className="box">
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                filter={filter}
+                handleChange={handleChangeQuery}
+                handleReset={handleQueryReset}
+                handleFilterChange={handleFilterChange}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!todos.length && (
+                <Loader />
+              )}
+              {todos.length && (
+                <TodoList
+                  todos={visibleTodos}
+                  selectedTodoId={selectedTodoId}
+                  changeTodo={changeTodo}
+                  changeUser={changeUser}
+                />
+              )}
             </div>
           </div>
+          {selectedUserId !== null && (
```

Comment:
```suggestion
          {!selectedUserId && (
```
Oleg meant to do it in this way